pystac_client/client.py

```diff
@@ -97,19 +87,49 @@ def open(cls, url=None, headers=None):
             raise TypeError(
                 "'url' must be specified or the 'STAC_URL' environment variable must be set.")
 
-        def read_text_method(url):
-            request = Request(url, headers=headers or {})
-            return pystac_client.stac_io.read_text_method(request)
-
-        old_read_text_method = STAC_IO.read_text_method
-        STAC_IO.read_text_method = read_text_method
-        try:
-            catalog = cls.from_file(url)
-        finally:
-            STAC_IO.read_text_method = old_read_text_method
-        catalog.headers = headers
+        stac_io = StacApiIO(headers=headers)
+
+        catalog = cls.from_file(url, stac_io)
         return catalog
 
+    @classmethod
+    def from_file(cls, href: str, stac_io: Optional[pystac.StacIO] = None) -> "STACObject":
```

Comment:
Is this still needed after stac-utils/pystac#451?

Reply:
Nope!  removing :smile: